x-pack/plugins/security_solution/scripts/endpoint/resolver_generator.ts

```diff
@@ -75,7 +83,94 @@ async function doIngestSetup(kibanaURL: string) {
 }
 
 async function main() {
-  const argv = yargs.help().options({
+  const argv = argvFromCLI();
+  await doIngestSetup(argv.kibana);
+
+  const clientOptions: ClientOptions = {
+    node: argv.node,
+  };
+
+  const client = new Client(clientOptions);
+  if (argv.delete) {
+    await deleteIndices(
+      [argv.eventIndex, argv.metadataIndex, argv.policyIndex, argv.alertIndex],
+      client
+    );
+  }
+
+  let seed = argv.seed;
+  if (!seed) {
+    seed = Math.random().toString();
+    console.log(`No seed supplied, using random seed: ${seed}`);
+  }
+
+  const startTime = Date.now();
+
+  const treeOptions: TreeOptions = {
+    ancestors: argv.ancestors,
+    generations: argv.generations,
+    children: argv.children,
+    relatedEvents: argv.relatedEvents,
+    relatedAlerts: argv.relatedAlerts,
+    percentWithRelated: argv.percentWithRelated,
+    percentTerminated: argv.percentTerminated,
+    alwaysGenMaxChildrenPerNode: argv.maxChildrenPerNode,
+    ancestryArraySize: argv.ancestryArraySize,
+  };
+
+  const {
+    numHosts,
+    numDocs,
+    metadataIndex,
+    policyIndex,
+    eventIndex,
+    alertIndex,
+    alertsPerHost,
+  } = argv;
+
+  const random = seedrandom(seed);
+
+  // For each host
+  for (let hostIndex = 0; hostIndex < numHosts; hostIndex++) {
+    // create a new generator instance
+    const generator = new EndpointDocGenerator(random);
+
+    // generate and index host and policy documents
+    await indexHostDocs(numDocs, client, metadataIndex, policyIndex, generator);
+
+    // build a resolver tree with related events and related alerts
+    const alertEventTree: GeneratedAlertTree = generator.generatedAlertTree(treeOptions);
+    console.log(
+      `host #${hostIndex}: created 1 alert: event.id: "${alertEventTree.alertEvent.event.id}"`
+    );
+
+    /**
+     * Index the ancestry and the alert event itself regardless of `alertsPerHost` limit.
+     * Otherwise the origin (aka trigger) alert event might not be indexed.
+     */
+    const alertEvents = [...alertEventTree.ancestry, alertEventTree.alertEvent];
+
+    // index additional events until `alertsPerHost` limit is reached
+    for (const event of alertEventTree.descendants) {
+      if (alertEvents.length === alertsPerHost) {
+        break;
+      }
+      alertEvents.push(event);
```

Comment:
this should continue using logic to chunk the descendants into groups before bulk indexing them, otherwise for large trees it can run out of memory - the descendant tree grows exponentially

Reply:
also the name is misleading, these are descendant events not alerts